pymongo/collection.py

```diff
@@ -2360,6 +2361,189 @@ def index_information(
+        models: List[SearchIndexModel],
+        session: Optional[ClientSession] = None,
+        comment: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Create multiple search indexes for the current collection.
+
+        :Parameters:
+          - `models`: A list of :class:`~pymongo.operations.SearchIndexModel` instances.
+          - `session` (optional): a :class:`~pymongo.client_session.ClientSession`.
+          - `comment` (optional): A user-provided comment to attach to this
+            command.
+          - `**kwargs` (optional): optional arguments to the createSearchIndexes
+            command (like maxTimeMS) can be passed as keyword arguments.
+
+        :Returns:
+            A list of the newly created search index names.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
+
+        .. versionadded:: 4.5
+        """
+        if comment is not None:
+            kwargs["comment"] = comment
+
+        def gen_indexes():
+            for index in models:
+                if not isinstance(index, SearchIndexModel):
+                    raise TypeError(
+                        f"{index!r} is not an instance of pymongo.operations.SearchIndexModel"
+                    )
+                yield index.document
+
+        cmd = SON([("createSearchIndexes", self.name), ("indexes", list(gen_indexes()))])
+        cmd.update(kwargs)
+
+        with self._socket_for_writes(session) as sock_info:
+            resp = self._command(
+                sock_info,
+                cmd,
+                read_preference=ReadPreference.PRIMARY,
+                codec_options=_UNICODE_REPLACE_CODEC_OPTIONS,
+            )
+            return [index["name"] for index in resp["indexesCreated"]]
+
+    def drop_search_index(
+        self,
+        name: str,
+        session: Optional[ClientSession] = None,
+        comment: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Delete a search index by index name.
+
+        :Parameters:
+          - `name`: The name of the search index to be deleted.
+          - `session` (optional): a
+            :class:`~pymongo.client_session.ClientSession`.
+          - `comment` (optional): A user-provided comment to attach to this
+            command.
+          - `**kwargs` (optional): optional arguments to the dropSearchIndexes
+            command (like maxTimeMS) can be passed as keyword arguments.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
+
+        .. versionadded:: 4.5
+        """
+        cmd = SON([("dropSearchIndex", self.__name), ("name", name)])
+        cmd.update(kwargs)
+        if comment is not None:
+            cmd["comment"] = comment
+        with self._socket_for_writes(session) as sock_info:
+            self._command(
+                sock_info,
+                cmd,
+                read_preference=ReadPreference.PRIMARY,
+                allowable_errors=["ns not found", 26],
```

Comment:
Should we use _UNICODE_REPLACE_CODEC_OPTIONS here and update_search_index too?

Reply:
Done